semantics/semBakeryScript.sml

```diff
@@ -0,0 +1,67 @@
+open preamble basis
+
+open astBakeryTheory
+
+val _ = new_theory "semBakery";
+
+val freeprocs_def = Define`
+  freeprocs (LTau p)           = {p}
+∧ freeprocs (LCom p1 v1 p2 v2) = {p1;p2}
+∧ freeprocs (LSel p1 b p2)     = {p1;p2}
+`;
+
+val sender_def = Define`
+  sender (LTau p)           = NONE
+∧ sender (LCom p1 v1 p2 v2) = SOME p1
+∧ sender (LSel p1 b p2)     = SOME p1
+`;
+
+val receiver_def = Define`
+  receiver (LTau p)           = NONE
+∧ receiver (LCom p1 v1 p2 v2) = SOME p2
+∧ receiver (LSel p1 b p2)     = SOME p2
+`;
+
+val (trans_rules,trans_ind,trans_cases) = Hol_reln `
+(∀s v1 p1 v2 p2 d c.
+    FLOOKUP s (v1,p1) = SOME d
+    ⇒ trans (s,Com p1 v1 p2 v2 c) (LCom p1 v2 p2 v2) (s |+ ((v2,p2),d),c))
+∧ (∀s p1 b p2 c.
+    trans (s,Sel p1 b p2 c) (LSel p1 v p2) (s,c))
```

Comment:
I don't really get what is happening here

Reply:
The state s is updated so that the binding of variable v2 at process p2 binds to d (which is the value of the sent variable v1 at p1). I figured some later compilation steps might be simpler if we use explicit binding environments rather than substitution, but that is up for grabs.